lib/liquid/template.rb

```diff
@@ -144,6 +144,11 @@ def render!(*args)
     def tokenize(source)
       source = source.source if source.respond_to?(:source)
       return [] if source.to_s.empty?
+
+      if source =~ WhiteSpaceTrimCheck
+          source = source.gsub(WhiteSpaceTrimTags, WhiteSpaceTrimReplace)
```

Comment:
Need to work something out. Currently the whitespace would be in the previous token so either need to bring the whitespace in to the current token which kills the performance of the tokenizer regex split or create logic that looks back and modifies the stack if it finds a whitespace control symbol.